velox/functions/prestosql/StringFunctions.h

```diff
@@ -494,4 +496,111 @@ struct LevenshteinDistanceFunction {
   }
 };
 
+/// Wrap the sbstemmer library and use its sb_stemmer_stem
+/// to get word stem
+class Stemmer {
+ private:
+  sb_stemmer* sbstemmer;
+  Stemmer(sb_stemmer* stemmer) : sbstemmer(stemmer) {}
+
+ public:
+  ~Stemmer() {
+    sb_stemmer_delete(sbstemmer);
+  }
+
+  /// Get a Stemmer from the the map stored in thread local storage
+  /// or create a new one if it doesn't exist. Return NULL if the
+  /// specified lang is not supported.
+  static Stemmer* getStemmer(const char* lang) {
+    thread_local std::map<std::string, Stemmer*> stemmers;
+    if (auto found = stemmers.find(lang); found != stemmers.end()) {
+      return found->second;
+    }
+    // Only support ASCII and UTF-8
+    if (auto sbstemmer = sb_stemmer_new(lang, "UTF_8")) {
+      auto rev = new Stemmer(sbstemmer);
+      stemmers[lang] = rev;
+      return rev;
+    }
+    return NULL;
+  }
+
+  /// Get the word stem or NULL if out of memory
+  const char* stem(const std::string& input) {
```

Comment:
Can we pass std::string_view or StringView instead of std::string to avoid unnecessary copy?

Reply:
line 598 is using std::string, so I pass std::string& here.